### ISSUE TITLE
Text values are truncated to 32767 bytes if edited directly in the table widget

### ISSUE THREAD

**Issue by mgrojo** [sqlitebrowser/sqlitebrowser#1281]
### Details for the issue

Text values are truncated to 32767 bytes if edited directly in the table widget.

### Useful extra information

Steps to reproduce:

1. Import a text file in the dock editor of more that 32767 bytes and click Apply.
2. Click inside the cell in the table widget. Wait the time to avoid a double click Make a second click to edit the value.
3. The value already appears truncated and selected. The only way to avoid the truncation once in this phase is to press ESC. Otherwise, the truncated value is updated in the database.

#### I'm opening this issue because:

- [ ] DB4S is crashing
- [x] DB4S has a bug
- [ ] DB4S needs a feature
- [ ] DB4S has another problem

#### I'm using DB4S on:

- [ ] Windows: ( _version:_ ___ )
- [x] Linux: ( _distro:_ Ubuntu 16.04)
- [ ] Mac OS: ( _version:_ ___ )
- [ ] Other: ___

#### I'm using DB4S version:

- [ ] 3.10.1
- [ ] 3.10.0
- [ ] 3.9.1
- [x] Other: _3.10.99 (Dec 28 2017)

#### I have also:

- [x] Tried out the latest nightly version: https://github.com/sqlitebrowser/sqlitebrowser#nightly-builds
- [x] Searched for an existing similar issue: https://github.com/sqlitebrowser/sqlitebrowser/issues?utf8=%E2%9C%93&q=is%3Aissue%20

**Comment by mgrojo** [comment]
I suppose this is a limit in the Qt widget, but haven't found any reference.

**Comment by MKleusberg** [comment]
Nice catch 👍 That's definitely a problem!

The limit is in the QLineEdit class which is used as an editor inside the table view: https://doc.qt.io/qt-5/qlineedit.html#maxLength-prop

I have just pushed a commit which increases the size limit (unfortunately there doesn't seem to be a way to disable it entirely) and which doesn't allow writing data back into the database if the increased limit is still passed. Does that work for you? :smile:

**Comment by mgrojo** [comment]
Perfect! This is working for me. I don't think it's an issue not being able to remove the maximum if the truncation problem is avoided. These long texts are better edited in the cell editor. In fact, I discovered this while playing around with SVG files for the XML mode editor. At first I didn't know what I've done to break the images :smile: In this way I also found that the double-click did only transfer the focus to the Text editor, and not to any other selected editor. I've fixed that in the xml branch.
